lib/srv/discovery/discovery_test.go

```diff
@@ -1937,44 +1923,40 @@ func TestDiscoveryDatabaseRemovingDiscoveryConfigs(t *testing.T) {
 
 		// Check for new resource in reconciler
 		expectDatabases := []types.Database{awsRDSDB}
-		select {
-		case <-waitForReconcile:
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
 			actualDatabases, err := tlsServer.Auth().GetDatabases(ctx)
-			require.NoError(t, err)
-			require.Empty(t, cmp.Diff(expectDatabases, actualDatabases,
+			if !assert.NoError(t, err) {
+				t.FailNow()
```

Comment:
Isn't `t.Fail()` the same as `require.XXX` as require is just a wrapper around `asserts` package with the extra `t.Fail()` call? Basically, it's the same thing as you did here.
I think you should just return as `assert` already fails the test.

Reply:
Done, thank you.